File: backend/models/ai_usage.go
package models

import "time"

// AIUsageLog stores per-call AI token usage for authenticated users.
type AIUsageLog struct {
	ID               string    `json:"id" gorm:"type:uuid;primaryKey;default:gen_random_uuid()"`
	UserID           string    `json:"user_id" gorm:"type:uuid;not null;index"`
	Feature          string    `json:"feature" gorm:"size:32;not null;index"`
	ModelSource      string    `json:"model_source" gorm:"size:16;not null;index"`
	Provider         string    `json:"provider" gorm:"size:32;not null;index"`
	PublicModelID    *string   `json:"public_model_id" gorm:"type:uuid;index"`
	Model            string    `json:"model" gorm:"size:128;not null;index"`
	PromptTokens     int       `json:"prompt_tokens" gorm:"default:0;not null"`
	CompletionTokens int       `json:"completion_tokens" gorm:"default:0;not null"`
	TotalTokens      int       `json:"total_tokens" gorm:"default:0;not null"`
	ReasoningTokens  int       `json:"reasoning_tokens" gorm:"default:0;not null"`
	CacheHitTokens   int       `json:"cache_hit_tokens" gorm:"default:0;not null"`
	CacheMissTokens  int       `json:"cache_miss_tokens" gorm:"default:0;not null"`
	UsageStatus      string    `json:"usage_status" gorm:"size:16;default:unknown;not null;index"`
	Success          bool      `json:"success" gorm:"default:false;not null;index"`
	ErrorMessage     string    `json:"error_message" gorm:"size:512;default:''"`
	LatencyMs        int64     `json:"latency_ms" gorm:"default:0;not null"`
	CreatedAt        time.Time `json:"created_at" gorm:"index"`
}

func (AIUsageLog) TableName() string {
	return "ai_usage_logs"
}

File: frontend/src/types/admin/aiUsage.ts
export interface AIUsageTotals {
  request_count: number;
  prompt_tokens: number;
  completion_tokens: number;
  total_tokens: number;
  reasoning_tokens: number;
  cache_hit_tokens: number;
  cache_miss_tokens: number;
}

export interface AIUsageBreakdown {
  key: string;
  label: string;
  request_count: number;
  prompt_tokens: number;
  completion_tokens: number;
  total_tokens: number;
  reasoning_tokens: number;
  cache_hit_tokens: number;
  cache_miss_tokens: number;
}

export interface AIUsageModelBreakdown {
  model: string;
  provider: string;
  request_count: number;
  prompt_tokens: number;
  completion_tokens: number;
  total_tokens: number;
  reasoning_tokens: number;
  cache_hit_tokens: number;
  cache_miss_tokens: number;
}

export interface AIUsageDailyTrend {
  date: string;
  provider: string;
  model: string;
  request_count: number;
  total_tokens: number;
}

export interface UserUsageItem {
  user_id: string;
  username: string;
  email: string;
  tokens: number;
  requests: number;
}

export interface AdminAIUsageResponse {
  today: AIUsageTotals;
  month: AIUsageTotals;
  total: AIUsageTotals;
  providers: AIUsageBreakdown[];
  models: AIUsageModelBreakdown[];
  daily_trend: AIUsageDailyTrend[];
  top_users: UserUsageItem[];
  month_label: string;
}

// User-specific AI usage detail
export interface UserAIUsageDetail {
  today: AIUsageTotals;
  month: AIUsageTotals;
  total: AIUsageTotals;
  providers: AIUsageBreakdown[];
  models: AIUsageModelBreakdown[];
  recent: AIUsageRecord[];
  recent_total: number;
  daily_trend: AIUsageDailyTrend[];
  month_label: string;
}

export interface AIUsageRecord {
  id: string;
  feature: string;
  model_source: string;
  provider: string;
  model: string;
  prompt_tokens: number;
  completion_tokens: number;
  total_tokens: number;
  reasoning_tokens: number;
  cache_hit_tokens: number;
  cache_miss_tokens: number;
  usage_status: string;
  success: boolean;
  latency_ms: number;
  created_at: string;
}

// --- Audit Log ---
